pip/pip-304.md

```diff
@@ -0,0 +1,85 @@
+# Motivation
+
+We have a scenario that uses the `org.apache.pulsar.broker.intercept.BrokerInterceptor#messageProduced`, and need to
```

Comment:
The method `BrokerInterceptor#onMessagePublish` already provided the `headersAndPayload`, you can retrieve the metadata from the `headersAndPayload`. Of course, I'm not saying we should not have message metadata in `BrokerInterceptor#messageProduced`, Just provide a way to retrieve the message metadata without new changes to Pulsar.

Reply:
I know the `BrokerInterceptor#onMessagePublish` method provides the `headersAndPayload`, at this point, the message has not persisted, I expect to retrieve the message metadata after the message is persisted, and can retrieve the ledgerId and entryId.

Reply:
But you can have a cache for metadata of the inflight messages, the key can be (topic, producerId, sequence) and the value is metadata, after the message is published, just retrieve it from the cache. This is a solution without waiting for a new feature version release because we should not cherry-pick any APIs changes to LTS version and patch release.